tests/conftest.py

```diff
@@ -1,19 +1,144 @@
 import pytest
+import sys
+import json
+import logging
+from pathlib import Path
+from dotenv import load_dotenv
+from typing import Generator, Any
 
+# DSI: Add project root to path for script imports
+project_root = Path(__file__).parent.parent
+sys.path.insert(0, str(project_root))
+
+
+# --- Modified: Existing and New Pytest Hooks ---
 
 def pytest_addoption(parser):
     parser.addoption("--runslow", action="store_true", default=False, help="run slow tests")
+    parser.addoption("--run-dsi-tests", action="store_true", default=False, help="run DSI integration tests")
 
 
 def pytest_configure(config):
     config.addinivalue_line("markers", "slow: mark test as slow to run")
+    config.addinivalue_line("markers", "dsi: mark test as part of the DSI framework")
+
+    if config.getoption("--run-dsi-tests"):
+        logging.basicConfig(
+            level=logging.INFO,
+            format='%(asctime)s - %(levelname)s - %(message)s',
+            datefmt='%Y-%m-%d %H:%M:%S',
+            force=True
+        )
+        logging.info("--- DSI: Loading environment variables ---")
+        try:
+            dotenv_path = project_root / '.env'
+            if load_dotenv(dotenv_path=dotenv_path, override=True):
+                logging.info(f"DSI: Successfully loaded environment variables from: {dotenv_path}")
+            else:
+                logging.warning(f"DSI: Could not find .env file at {dotenv_path}. Using system variables.")
+        except Exception as e:
```

Comment:
Don't do this. We need to either catch exceptions we want to do something about, or let them raise